Origin/feature/implement connectivity checking for retailer list

Kwang-Kian-Hui opened the pull request:
Connectivity checking for retailer list:
1) when attempting to get retailer list
2) listen for connectivity changes while in consumer home page(retailer list)

used connectivity package as internet connectivity checker package does not allow listening of device connectivity.

Kwang-Kian-Hui committed:
build_runner for firestore_failures

added no connection failure

Kwang-Kian-Hui committed:
implement connectivity checker feature

import connectivity package v^3.0.6.
snackbar to notify user of "no connection" when no connection detected

Kwang-Kian-Hui committed:
clean up unnecessary comments

Kwang-Kian-Hui committed:
Merge branch 'main' into origin/feature/implement-connectivity-checking-for-retailer-list

sinhaugoh commented on lib/consumer/presentation/consumer_home_page.dart:
```diff
@@ -1,29 +1,56 @@
+    connectivityChecker();
     final mq = MediaQuery.of(context);
     return Scaffold(
       appBar: enterLocationAppBar(),
```
I would suggest u put your listener in initstate.

case 1: The user's device has no connection when logged in
1. The listener will not perform any action when the device is not connected.
2. The notifier check and return a newly created RetailerListState.noConnection() instead of failure
3. In the presentation, when the state is noConnection, display a text in the scaffold telling the user that they are not connected

4. when the user connects to the internet, the getRetailerList will get triggered in the listener and the listener will be disposed right after the method is called.

What u think?

Kwang-Kian-Hui replied:
okay i'll try that

Kwang-Kian-Hui committed:
changed noConnection from firestore failure to state

removed noConnection firestore failure and switched to state.
displays text when state is noConnection

Kwang-Kian-Hui committed:
feature add on - refresh page

Used refresh indicator(pull down refresh) to allow refresh of consumer home page

Kwang-Kian-Hui committed:
Merge branch 'main' into origin/feature/implement-connectivity-checking-for-retailer-list

sinhaugoh committed:
adjust no connection flow for consumer home page
